examples/recv.py

```diff
@@ -38,9 +38,9 @@ def on_event_data(self, event_data):
 
 try:
     ADDRESS = ("amqps://"
-               "<URL-encoded-SAS-policy>"
+               "<SAS-policy>"
```

Comment:
The policy name and key need to be URL encoded as they may contain ':', '@' or '/' that can cause parsing errors.

Reply:
coming from azure they're already encoded, the /'s and +'s in my example code are examples of items included in a key I already use. When I do a urlencode with urllib, it re-encodes those, and breaks the connection.

For a more specific example, I've literally taken a key that I copied from the azure UI, and replaced the non-special (letters and numbers) characters in the string:
PviviviQRPZ55pJ+rr1wQWERTffDmrD1llPDDj/3RuMN=
Note, the key I get FROM AZURE already has the + / and = in the key.

```python
>>> from urllib.parse import urlencode
>>> encoded_key = urlencode({'SharedAccessKey': 'PviviviQRPZ55pJ+rr1wQWERTffDmrD1llPDDj/3RuMN='})
>>> print(encoded_key)
SharedAccessKey=PviviviQRPZ55pJ%2Brr1wQWERTffDmrD1llPDDj%2F3RuMN%3D
```

the output now encodes the + as %2B, the / as %2F and the = as %3D

if I pass those into the connection string as:
"amqps://KeyName:PviviviQRPZ55pJ%2Brr1wQWERTffDmrD1llPDDj%2F3RuMN%3D@mynamespace.servicebus.windows.net/myeventhub"

Then I get permissions errors.

However, if I pass them in as:
"amqps://KeyName:PviviviQRPZ55pJ+rr1wQWERTffDmrD1llPDDj/3RuMN=@mynamespace.servicebus.windows.net/myeventhub"

I connect fine.

Reply:
This is interesting. The key you get from Azure portal is BASE64 encoded which can contain '+' and '/'. That's why it needs to be URL encoded again as the password in the address Url. When I tried the non-encoded address string, I am getting a Url parsing error from Proton which is expected.
```
C:\Users\mqt\repo\azure-event-hubs-python>python examples\send.py
Traceback (most recent call last):
  File "C:\Python36\lib\proton\__init__.py", line 4203, in _port_int
    return int(value)
ValueError: invalid literal for int() with base 10: 'PviviviQRPZ55pJ+rr1wQWERTffDmrD1llPDDj'
```